Mining on MacBookPro issue with cpp-ethereum

Hi All,

Running the following command failed:

$ brew reinstall cpp-ethereum --with-gpu-mining --devel --headless --build-from-source
Warning: Calling build.build_bottle? is deprecated!
Use build.bottle? instead.
/usr/local/Homebrew/Library/Homebrew/cmd/reinstall.rb:37:in `reinstall_formula'

==> Reinstalling ethereum/ethereum/cpp-ethereum 
Warning: ethereum/ethereum/cpp-ethereum: this formula has no --headless option so it will be ignored!
Warning: ethereum/ethereum/cpp-ethereum: this formula has no --with-gpu-mining option so it will be ignored!
==> Cloning https://github.com/ethereum/cpp-ethereum.git
Updating /Users/romanmagda/Library/Caches/Homebrew/cpp-ethereum--git
==> Checking out branch develop
==> cmake -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -D
Last 15 lines from /Users/romanmagda/Library/Logs/Homebrew/cpp-ethereum/01.cmake:
CMake Error in test/CMakeLists.txt:
  Target "web3jsonrpc" contains relative path in its
  INTERFACE_INCLUDE_DIRECTORIES:

    "MHD_INCLUDE_DIR-NOTFOUND"


-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    HAVE_CLOCK_GETTIME


-- Build files have been written to: /tmp/cpp-ethereum-20170526-16385-1f7a4yx

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/ethereum/homebrew-ethereum/issues

Please help me to solve the issue, thank you very much !

Roman

--- a/README.md
+++ b/README.md
@@ -120,21 +120,6 @@ bottles directly with `brew install <bottle_url>`, see [cpt-obvious](https://bui
 for previous builds.
 
 
-## Options
-
-See `brew info ethereum` or `brew info cpp-ethereum` for all options. `--with-...` features are experimental patches.
-
-#### cpp-ethereum options
-Option                 | desc.
------------------------|---------
-`--with-evmjit`        | Build with LLVM and enable EVMJIT
-`--successful`         | Last successful build using --devel only
-`--with-debug`         | Pass -DCMAKE_BUILD_TYPE=Debug
-`--with-vmtrace`       | Build with -DVMTRACE=1
-`--with-paranoia`      | Build with -DPARANOID=1
-
-**Note:** `--with-evmjit` requires LLVM to be installed with `brew install llvm --HEAD --with-clang`
-
 ## Troubleshooting
 
 * Use `--verbose` to get more info while installing.

--- a/cpp-ethereum.rb
+++ b/cpp-ethereum.rb
@@ -47,6 +47,9 @@ class CppEthereum < Formula
     system "cmake", *args
     system "make"
     system "make", "install"
+  end
 
+  test do
+    system "#{bin}/eth", "--version"
   end
 end

--- a/cpp-ethereum.rb
+++ b/cpp-ethereum.rb
@@ -13,8 +13,6 @@
 # (c) 2014-2017 cpp-ethereum contributors.
 #------------------------------------------------------------------------------
 
-require 'formula'
-
 class CppEthereum < Formula
   version '1.3.0'
 

--- a/cpp-ethereum.rb
+++ b/cpp-ethereum.rb
@@ -19,32 +19,14 @@ class CppEthereum < Formula
   url "https://github.com/ethereum/cpp-ethereum.git", :branch => "develop"
   version "1.3.0-develop"
 
-  option "without-evmjit", "Build without JIT (and its LLVM dependency)"
-  option "with-debug", "Build with debug"
-  option "with-vmtrace", "Build with VMTRACE"
-  option "successful", "Last successful build with --devel only"
-
   depends_on "cmake" => :build
   depends_on "leveldb"
 
   def install
     args = *std_cmake_args
+    args << "-DVMTRACE=ON"
 
-    if build.with? "debug" or build.with? "vmtrace" or build.with? "paranoia"
-      args << "-DCMAKE_BUILD_TYPE=Debug"
-    else
-      args << "-DCMAKE_BUILD_TYPE=Release"
-    end
-
-    if build.without? "evmjit"
-      args << "-DEVMJIT=0"
-    else
-      args << "-DEVMJIT=1"
-    end
-
-    args << "-DVMTRACE=1" if build.with? "vmtrace"
-
-    system "cmake", *args
+    system "cmake", ".", *args
     system "make"
     system "make", "install"
   end

--- a/cpp-ethereum.rb
+++ b/cpp-ethereum.rb
@@ -24,13 +24,8 @@ class CppEthereum < Formula
   option "with-vmtrace", "Build with VMTRACE"
   option "successful", "Last successful build with --devel only"
 
-  depends_on 'boost'
-  depends_on 'cmake' => :build
-  depends_on 'curl'
-  depends_on 'gmp'
-  depends_on 'leveldb'
-  depends_on 'libjson-rpc-cpp'
-  depends_on 'miniupnpc'
+  depends_on "cmake" => :build
+  depends_on "leveldb"
 
   def install
     args = *std_cmake_args

--- a/cpp-ethereum.rb
+++ b/cpp-ethereum.rb
@@ -65,31 +65,4 @@ class CppEthereum < Formula
     system "make", "install"
 
   end
-
-  def plist; <<~EOS
-    <?xml version="1.0" encoding="UTF-8"?>
-    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-    <plist version="1.0">
-      <dict>
-        <key>Label</key>
-        <string>#{plist_name}</string>
-        <key>RunAtLoad</key>
-        <true/>
-        <key>KeepAlive</key>
-        <true/>
-        <key>ThrottleInterval</key>
-        <integer>300</integer>
-        <key>ProgramArguments</key>
-        <array>
-            <string>#{opt_bin}/eth</string>
-            <string>-d</string>
-            <string>#{prefix}/.ethereum</string>
-            <string>-b</string>
-        </array>
-        <key>WorkingDirectory</key>
-        <string>#{HOMEBREW_PREFIX}</string>
-      </dict>
-    </plist>
-    EOS
-  end
 end

--- a/cpp-ethereum.rb
+++ b/cpp-ethereum.rb
@@ -14,21 +14,10 @@
 #------------------------------------------------------------------------------
 
 class CppEthereum < Formula
-  version '1.3.0'
-
-  homepage 'http://cpp-ethereum.org'
-  url 'https://github.com/ethereum/cpp-ethereum.git', :branch => 'develop'
-
-  devel do
-
-    if build.include? "successful"
-      version '1.3.0'
-      url 'https://github.com/ethereum/cpp-ethereum.git', :revision => '4943d3eeae865537e18c410e5e7d064825bf2711'
-    else
-      version '1.3.0'
-      url 'https://github.com/ethereum/cpp-ethereum.git', :branch => 'develop'
-    end
-  end
+  desc "Ethereum C++ client"
+  homepage "http://cpp-ethereum.org"
+  url "https://github.com/ethereum/cpp-ethereum.git", :branch => "develop"
+  version "1.3.0-develop"
 
   option "without-evmjit", "Build without JIT (and its LLVM dependency)"
   option "with-debug", "Build with debug"

--- a/ethereum.rb
+++ b/ethereum.rb
@@ -1,5 +1,3 @@
-require 'formula'
-
 class Ethereum < Formula
   homepage 'https://github.com/ethereum/go-ethereum'
   url 'https://github.com/ethereum/go-ethereum.git', :tag => 'v1.8.2'

--- a/solidity.rb
+++ b/solidity.rb
@@ -13,8 +13,6 @@
 # (c) 2014-2017 solidity contributors.
 #------------------------------------------------------------------------------
 
-require 'formula'
-
 class Solidity < Formula
 
   desc "The Solidity Contract-Oriented Programming Language"

